File: station-finder-backend/src/main/java/com/ganesh/finder/dto/StationWithScore.java
package com.ganesh.finder.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class StationWithScore {
    private Long id;
    private String name;
    private Double latitude;
    private Double longitude;
    private String address;
    private String operatingHours;
    private Double pricePerKwh;
    private Double rating;
    private Boolean isOpen;
    private String meta;

    // Computed fields
    private Double distance;
    private Integer availableSlots;
    private Integer totalSlots;
    private List<String> connectorTypes;
    private List<SlotInfo> slots;

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class SlotInfo {
        private Long id;
        private String label;
        private String connectorType;
        private Double powerKw;
        private Boolean isAvailable;
    }
}

File: android/app/src/main/java/com/ganesh/ev/data/model/StationWithScore.kt
package com.ganesh.ev.data.model

import java.io.Serializable

data class StationWithScore(
        val station: Station,
        val distance: Double,
        val score: Double,
        val trafficScore: Double,
        val gridScore: Double,
        val parkingScore: Double,
        val accessScore: Double,
        val availableSlots: Int,
        val totalSlots: Int,
        val connectorTypes: List<String> = emptyList(),
        val lastActive: String = "Unknown",
        val rating: Double = 0.0
) : Serializable
